server/src/sim/financialanalysis.py

```diff
@@ -7,8 +7,9 @@
 import copy
 from setoptions import setoptions
 from copy import deepcopy
+from utils import sanitize
 
-def financialanalysis(D, postyear=2015, S=None, yscale='abs', makeplot=False):
+def financialanalysis(D, postyear=2015.0, S=None, yscale='abs', makeplot=False):
```

Comment:
That will not help if financialanalysis is still called with an integer. Better to convert an argument to a float inside the function.